### ISSUE TITLE
Multiplayer On kicked event is not triggered and Multiplayer.PeerCount is not updated

### ISSUE THREAD

**Issue by esowe** [Scirra/Construct-3-bugs#1902]
## Problem description
Because of another bug ([https://github.com/Scirra/Construct-3-bugs/issues/1901](url)) I haven't been able to test this on the Construct 2 runtime. However, the first time I discovered this bug some time ago, I was using the old runtime, so most likely it is also affected.

When a multiplayer host manually kicks a peer (using the Kick peer action of the multiplayer plugin) the peer is kicked as expected, but the On kicked event isn't triggered on that peer.

Possibly related to this, if a peer is kicked by the host as described above, the remaining peers aren't properly notified. Multiplayer.PeerCount increases when a peer connects, but it doesn't decrease when one is kicked. Also, its value is not the same on all peers.

If a peer is kicked as a result of the host leaving, the On kicked event is properly triggered.

## Attach a .c3p
[Multiplayer On kicked and peercount.zip](https://github.com/Scirra/Construct-3-bugs/files/2264335/Multiplayer.On.kicked.and.peercount.zip)

## Steps to reproduce
1. Open a remote preview of the project and copy the remote preview URL.
2. Open a new window in Google Chrome.
3. Paste the remote preview URL in the tab and open it.
4. Open two more tabs in Google Chrome and repeat step 3 for all tabs.
5. Go to the first tab and wait until it has connected to the signalling server.
6. When the topmost button changes text to "Online - Login", push it. This tab will then become the host.
7. Change to the second tab and repeat steps five and six above. This tab will then become the first peer. Note that both the new peer and the host states that there are two peers, which is correct (one is the host).
8. Go back to the first tab and press "Lock game". This will immediately kick any new peer that tries to connect.
9. Go to the third tab and press the "Online - Login" button. The peer is not allowed to join. The host has kicked the peer (see the first tab), but the On kicked event (event number 23) is never triggered on the peer. Also note that the host claims that there are still three peers, although one has just been kicked.
10. On the third tab, press the "Online - Login" button again three more times. Each time, the host kicks the peer on the third tab, but the number of peers connected (as given by Multiplayer.PeerCount) increases by one on the host and the connected peer on tab two. No On kicked event is triggered on the kicked peer. After this step, both the host and the peer on tab two thinks that the number of peers (including the host itself) is six, although in reality it is only two.
11. Go to the first tab and press "Unlock game". This will stop kicking peers trying to connect.
12. Go to the third tab and press the "Online - Login" button. This peer will now be allowed to join.
13. Both the host and peer on the second tab now think that seven peers are connected. The peer on the third tab thinks that fifteen peers are connected.
14. If needed, press the "Clear Debug Text" button on the third tab to make sure that there is room for more text.
15. Go to the first tab (the host) and press "Logged in - Log out". This will disconnect the host. Note that both peers on tabs two and three are now properly kicked and that the On kicked event does trigger on both peers.

## Observed result

Firstly, the Multiplayer On kicked event is not triggered on the kicked peer if the host manually kicks that peer as described above.
Secondly, and possibly related to the first observation, the value of Multiplayer.PeerCount is incorrect both on the host and all connected peers if the host manually kicks peers.
Thirdly, the value of Multiplayer.PeerCount is not only incorrect, but it is also not the same on all connected peers all the time.

## Expected result

I expect that the On kicked event is triggered whenever the host manually kicks a peer and not only when the host disconnects.
I expect that Multiplayer.PeerCount is updated whenever a peer is kicked and that it always (after a reasonable delay) reflects the number of connected peers (including the host, as described in the manual)
I expect that Multiplayer.PeerCount is the same on all connected peers.

## Affected browsers

* Chrome: yes
* Firefox: unknown
* Edge: unknown
* Safari: unknown

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 67.0.3396.99
Browser engine: Blink
Browser architecture: 64-bit
Context: webapp
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 4
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Safari/537.36
C3 release: r111 (stable)
Language setting: en-US

Local storage
Storage quota (approx): 32 gb
Storage usage (approx): 165 mb (0.5%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Vendor: Google Inc.
Renderer: ANGLE (Intel(R) HD Graphics 620 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
Should have both fixed for the next release ('kick peer' not triggering 'on kicked' at remote end; C3 runtime not tracking peer counts correctly).

**Comment by esowe** [comment]
I can confirm that these bugs have been fixed for the Construct 3 runtime in the R112 release. Thanks!

However, since error 1901 (https://github.com/Scirra/Construct-3-bugs/issues/1901) has stopped occurring after the update to R112, I have now been able to test this for the old Construct 2 runtime too, and both the on kicked event bug and the PeerCount bug still occur when the Construct 2 runtime is used.
